go/tasks/plugins/array/k8s/launcher.go

```diff
@@ -68,113 +59,25 @@ func applyPodTolerations(_ context.Context, cfg *Config, pod *corev1.Pod) *corev
-	}
-
-	podTemplate, _, err := FlyteArrayJobToK8sPodTemplate(ctx, tCtx)
-	if err != nil {
-		return currentState, errors2.Wrapf(ErrBuildPodTemplate, err, "Failed to convert task template to a pod template for task")
-	}
-
-	var args []string
-	if len(podTemplate.Spec.Containers) > 0 {
-		args = append(podTemplate.Spec.Containers[0].Command, podTemplate.Spec.Containers[0].Args...)
-		podTemplate.Spec.Containers[0].Command = []string{}
-	}
-
-	size := currentState.GetExecutionArraySize()
-	// TODO: Respect parallelism param
-	for i := 0; i < size; i++ {
-		pod := podTemplate.DeepCopy()
-		indexStr := strconv.Itoa(i)
-		pod.Name = formatSubTaskName(ctx, tCtx.TaskExecutionMetadata().GetTaskExecutionID().GetGeneratedName(), indexStr)
-		pod.Spec.Containers[0].Env = append(pod.Spec.Containers[0].Env, corev1.EnvVar{
-			Name:  FlyteK8sArrayIndexVarName,
-			Value: indexStr,
-		})
-
-		pod.Spec.Containers[0].Env = append(pod.Spec.Containers[0].Env, arrayJobEnvVars...)
-
-		pod.Spec.Containers[0].Args, err = utils.ReplaceTemplateCommandArgs(ctx, args, arrayJobInputReader{tCtx.InputReader()}, tCtx.OutputWriter())
-		if err != nil {
-			return currentState, errors2.Wrapf(ErrReplaceCmdTemplate, err, "Failed to replace cmd args")
-		}
-
-		pod = ApplyPodPolicies(ctx, config, pod)
-		pod = applyNodeSelectorLabels(ctx, config, pod)
-		pod = applyPodTolerations(ctx, config, pod)
-
-		err = kubeClient.GetClient().Create(ctx, pod)
-		if err != nil && !k8serrors.IsAlreadyExists(err) {
-			if k8serrors.IsForbidden(err) {
-				if strings.Contains(err.Error(), "exceeded quota") {
-					// TODO: Quota errors are retried forever, it would be good to have support for backoff strategy.
-					logger.Infof(ctx, "Failed to launch job, resource quota exceeded. Err: %v", err)
-					currentState = currentState.SetPhase(arrayCore.PhaseWaitingForResources, 0).SetReason("Not enough resources to launch job.")
-				} else {
-					currentState = currentState.SetPhase(arrayCore.PhaseRetryableFailure, 0).SetReason("Failed to launch job.")
-				}
-
-				currentState = currentState.SetReason(err.Error())
-				return currentState, nil
-			}
-
-			return currentState, errors2.Wrapf(ErrSubmitJob, err, "Failed to submit job")
-		}
-	}
-
-	logger.Infof(ctx, "Successfully submitted Job(s) with Prefix:[%v], Count:[%v]", tCtx.TaskExecutionMetadata().GetTaskExecutionID().GetGeneratedName(), size)
-
-	arrayStatus := arraystatus2.ArrayStatus{
-		Summary:  arraystatus2.ArraySummary{},
-		Detailed: arrayCore.NewPhasesCompactArray(uint(size)),
-	}
-
-	currentState.SetPhase(arrayCore.PhaseCheckingSubTaskExecutions, 0).SetReason("Job launched.")
-	currentState.SetArrayStatus(arrayStatus)
-
-	return currentState, nil
-}
-
-func TerminateSubTasks(ctx context.Context, tMeta core.TaskExecutionMetadata, kubeClient core.KubeClient,
-	errsMaxLength int, currentState *arrayCore.State) error {
+func TerminateSubTasks(ctx context.Context, tCtx core.TaskExecutionContext, kubeClient core.KubeClient, config *Config,
```

Comment:
I would either rename it to FinalizeSubTasks or change it to call Abort & Finalize instead

Reply:
I updated it to call Abort & Finalize.